frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -54,4 +58,32 @@ export function getField(activity, fieldName, defaultValue){
     return defaultValue;
   }
   return activity[fieldName];
-}
\ No newline at end of file
+}
+
+/**
+ * Write contents into an activity in the database.
+ * 
+ * @param {string} tripId Database ID of the trip whose actiivty should be modified.
+ * @param {string} activityId Database ID of the activity to be modified.
+ * @param {Object} newValues Dictionary of the new values in {fieldName: newValue} form
+ * @returns true if the write was successful, false otherwise.
+ */
+export async function writeActivity(tripId, activityId, newValues) {
+  // todo: check if tripId or activityId is not valid. (#58)
+  newValues = {
+    ...newValues, 
+    'fillerstamp': Firebase.firestore.Timestamp.now()
+  };
+
+  const activity = db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES).doc(activityId);
+  
+  try {
+    // Note: newValues cannot contain lists. Check documentation for update().
+    const a = await act.update(newValues);
```

Comment:
Did you see what `a` was in either case (did or did not exist) because I think it may be undefined in either case given that [update](https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference#update) returns a void promise (I could be wrong but I found [this](https://stackoverflow.com/questions/44216949/does-a-async-function-which-returns-promisevoid-have-an-implicit-return-at-the) when trying to confirm my suspicions).

Reply:
I found something contradictory, but I can't find it anymore. Fixed.